[Regression] - Set value as string when doing a strict comparison.

tonypartridge opened the pull request:
Pull Request for Issue #18054 #17975 #18039 

Was caused partially by: 
https://github.com/joomla/joomla-cms/pull/12220


### Summary of Changes
Force to string when comparing against a string. 

@wilsonge - Test please :p

tonypartridge committed:
Set value as string when doing a strict comparision.

brianteeman commented:
did you mean to change the code in the simplepie file or is that a search/replace error? That is a vendor supplied file and we wouldnt usually touch those files

tonypartridge committed:
Revert vendor file change.

tonypartridge commented:
@brianteeman I added it as an to be safe as wasn't 100% sure it was a simpleXML element. I didn't check the full path so didn't notice it was vendor specific. I've reverted that one now, thanks!

infograf768 commented:
Part of this is already in https://issues.joomla.org/tracker/joomla-cms/18054 (which I tested OK).

Maybe better close #18054 ?

tonypartridge commented:
Whoops! I closed the issue not the pull too, thanks. I’ve now closed that
pull.

-- 
Tony Partridge

On 22 September 2017 at 10:32:32, infograf768 (notifications@github.com)
wrote:

Part of this is already in
https://issues.joomla.org/tracker/joomla-cms/18054 (which I tested OK).

Maybe better close #18054 <https://github.com/joomla/joomla-cms/pull/18054>
?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/joomla/joomla-cms/pull/18061#issuecomment-331399397>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABVglppOcyqm6yH3cQgNcEI0Jq41Uhx8ks5sk36wgaJpZM4PgaPH>
.

wilsonge commented on libraries/src/Form/Field/TagField.php:
```diff
@@ -243,7 +243,7 @@ public function isNested()
 		{
 			// If mode="nested" || ( mode not set & config = nested )
-			if (isset($this->element['mode']) && (string) $this->element['mode'] === 'nested'
+			if (isset($this->element['mode']) && (string) (string) $this->element['mode'] === 'nested'
```
You have two string typehints here ;)

tonypartridge committed:
Removed duplicated (String)

tonypartridge commented:
No Comment… updated.

-- 
Tony Partridge

On 22 September 2017 at 11:16:29, George Wilson (notifications@github.com)
wrote:

*@wilsonge* commented on this pull request.
------------------------------

In libraries/src/Form/Field/TagField.php
<https://github.com/joomla/joomla-cms/pull/18061#discussion_r140458745>:

> @@ -243,7 +243,7 @@ public function isNested()
                if ($this->isNested === null)
                {
                        // If mode="nested" || ( mode not set & config
= nested )
-                       if (isset($this->element['mode']) && (string)
$this->element['mode'] === 'nested'
+                       if (isset($this->element['mode']) && (string)
(string) $this->element['mode'] === 'nested'

You have two string typehints here ;)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/joomla/joomla-cms/pull/18061#pullrequestreview-64536586>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABVglkp5cOCo92orH-Rw6KUnwsfAzW39ks5sk4j9gaJpZM4PgaPH>
.

infograf768 commented:
I have tested this item :white_check_mark: successfully on fa86cfc1561a18fb8f02975e6ba5c94946acb0f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18061">issues.joomla.org/tracker/joomla-cms/18061</a>.</sub>

infograf768 commented:
We need a second tester here.

ghost commented:
@drmenzelit can you please test?

alikon commented:
I have tested this item :white_check_mark: successfully on fa86cfc1561a18fb8f02975e6ba5c94946acb0f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18061">issues.joomla.org/tracker/joomla-cms/18061</a>.</sub>

ghost commented:
RTC after two successful tests.

wilsonge commented:
Thanks Tony!
